packages/dbos-cloud/applications/deploy-app-code.ts

```diff
@@ -145,21 +147,37 @@ export async function deployAppCode(
     } else {
       logger.info("Time travel is disabled for this application");
     }
-    const ret = await registerApp(userDBName, host, enableTimeTravel, appName);
+    const ret = await registerApp(userDBName, host, enableTimeTravel, appName, executorsMemoryMib);
     if (ret !== 0) {
       return 1;
     }
   } else {
     logger.info(`Application ${appName} exists, updating...`);
     if (userDBName && appRegistered.PostgresInstanceName !== userDBName) {
-      logger.warn(`Application ${chalk.bold(appName)} is deployed with database instance ${chalk.bold(appRegistered.PostgresInstanceName)}. Ignoring the provided database instance name ${chalk.bold(userDBName)}.`);
+      logger.warn(
+        `Application ${chalk.bold(appName)} is deployed with database instance ${chalk.bold(appRegistered.PostgresInstanceName)}. Ignoring the provided database instance name ${chalk.bold(
+          userDBName
+        )}.`
+      );
     }
 
     // Make sure the app database is the same.
-    if (appRegistered.ApplicationDatabaseName && (dbosConfig.database.app_db_name !== appRegistered.ApplicationDatabaseName)) {
-      logger.error(`Application ${chalk.bold(appName)} is deployed with app_db_name ${chalk.bold(appRegistered.ApplicationDatabaseName)}, but ${dbosConfigFilePath} specifies ${chalk.bold(dbosConfig.database.app_db_name)}. Please update the app_db_name field in ${dbosConfigFilePath} to match the database name.`);
+    if (appRegistered.ApplicationDatabaseName && dbosConfig.database.app_db_name !== appRegistered.ApplicationDatabaseName) {
+      logger.error(
+        `Application ${chalk.bold(appName)} is deployed with app_db_name ${chalk.bold(appRegistered.ApplicationDatabaseName)}, but ${dbosConfigFilePath} specifies ${chalk.bold(
+          dbosConfig.database.app_db_name
+        )}. Please update the app_db_name field in ${dbosConfigFilePath} to match the database name.`
+      );
       return 1;
     }
+
+    // Check if the user wants to update the executors memory
+    if (executorsMemoryMib && appRegistered.ExecutorsMemoryMib !== executorsMemoryMib) {
+      const ret = await updateApp(host, appName, executorsMemoryMib, userCredentials);
+      if (ret !== 0) {
+        return 1;
+      }
+    }
```

Comment:
If the users provides a desired RAM amount, and this amount is different from what the application record has, then update the application record.